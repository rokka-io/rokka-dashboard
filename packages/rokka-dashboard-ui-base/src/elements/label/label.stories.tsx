import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';

import { Label } from './label';

storiesOf('Elements / Forms', module).add('Label', () => <Label>{text('text', 'A label')}</Label>);
