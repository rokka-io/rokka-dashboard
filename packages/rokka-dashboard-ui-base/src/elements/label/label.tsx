import styled from 'styled-components';
import { colors } from '../../identity/colors/colors';
import { fonts } from '../../identity/typography/index';
import { spaces } from '../../identity/spaces/spaces';

export const Label = styled.label`
  font-family: ${fonts.families.brand};
  font-size: ${fonts.sizes.xsmall};
  color: ${colors.gray.darkest};
  font-weight: ${fonts.weights.bold};
  margin-bottom: ${spaces.small};
`;
