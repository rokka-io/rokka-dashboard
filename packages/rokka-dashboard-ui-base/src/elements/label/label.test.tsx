import { shallow } from 'enzyme';
import React from 'react';

import { Label } from './label';

describe('Elements', () => {
  describe('Label', () => {
    it('renders a label correctly.', () => {
      const component = shallow(<Label>test</Label>);

      expect(component).toMatchSnapshot();
    });
  });
});
