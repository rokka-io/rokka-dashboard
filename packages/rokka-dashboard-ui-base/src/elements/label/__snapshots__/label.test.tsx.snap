// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements Label renders a label correctly. 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  font-family: ",
          "\\"Liip Etica Book\\", sans-serif",
          ";
  font-size: ",
          ".75rem",
          ";
  color: ",
          "#898A81",
          ";
  font-weight: ",
          "700",
          ";
",
        ],
      },
      "displayName": "styled.label",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "label",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  test
</StyledComponent>
`;
